beets/dbcore/query.py

```diff
@@ -553,16 +553,16 @@ def parse(cls, string):
         """
         if not string:
             return None
-        ordinal = string.count('-')
-        if ordinal >= len(cls.date_formats):
-            # Too many components.
-            raise InvalidQueryArgumentTypeError(string,
-                                                'a valid datetime string')
-        date_format = cls.date_formats[ordinal]
-        try:
-            date = datetime.strptime(string, date_format)
-        except ValueError:
-            # Parsing failed.
+        ordinal = 0
+        date = None
+        for date_format in cls.date_formats:
+            try:
+                date = datetime.strptime(string, date_format)
+                break
+            except ValueError:
+                # Parsing failed.
+                ordinal += 1
```

Comment:
What do you think about using

```
for ordinal, date_format in enumerate(cls.date_formats):
```

instead of manually incrementing `ordinal`? If that can work, it might be a little more readable.